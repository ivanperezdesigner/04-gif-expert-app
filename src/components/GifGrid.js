import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem.js'
import { getGift } from '../helpers/getGifs.js'

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGift(category)
            .then(imgs => setImages( imgs))
    }, [category])

    return (
        <>
            <h3>{ category }</h3>
            <div className='card-grid'>
                {
                    images.map(img => (
                    <GifGridItem 
                        key={img.id}
                        {...img}
                    />
                    ))
                }
            </div>
        </>
    )
}
