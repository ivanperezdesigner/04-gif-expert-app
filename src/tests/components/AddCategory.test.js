import { shallow } from 'enzyme'
import { AddCategory } from '../../components/AddCategory'
import React from 'react'

describe('Pruebas en el componente', () => {
    const setCategories = () => {}
    const wrapper = shallow(<AddCategory setCategories={ setCategories } />)
    
    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('evaluar los cambios en la caja de texto', () => {
        const input = wrapper.find('input')
        const value = 'Hola mundo'
        input.simulate('change', {target: {value}})
        expect (wrapper.find('p').text().trim()).toBe(value)
    })
})
