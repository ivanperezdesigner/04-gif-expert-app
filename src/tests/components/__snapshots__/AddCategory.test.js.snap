// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
